framework/auth/cas.py

```diff
@@ -257,14 +257,13 @@ def make_response_from_ticket(ticket, service_url):
                 cas_resp.attributes['accessToken'],
                 redirect(service_furl.url)
             )
-        # first time login from remote oauth provider, user does not exist or is not linked
-        if not user and action == 'external_first_login':
+        # first time login from external identity provider
+        if not user and external_credential and action == 'external_first_login':
             from website.util import web_url_for
-            # TODO: when cas part is ready, change the attribute names accordingly
             user = {
-                'fullname': cas_resp.attributes['fullname'],
-                'external_id_provider': cas_resp.attributes['oauthProvider'],
-                'external_id': cas_resp.attributes['oauthId'],
+                'external_id_provider': external_credential['provider'],
+                'external_id': external_credential['id'],
+                'fullname': cas_resp.attributes['given-names'] + ' ' + cas_resp.attributes['family-name'],
```

Comment:
Very good case. I am not sure. Need to take a look at OAuth specs. It is safe for ORCID for now, but we may want to handle this in CAS or OSF when we have other external IDPs. I will make a TODO here.